internal/transport/http2_client.go

```diff
@@ -1322,7 +1327,8 @@ func (t *http2Client) reader() {
 	}
 	sf, ok := frame.(*http2.SettingsFrame)
 	if !ok {
-		t.Close() // this kicks off resetTransport, so must be last before return
+		// this kicks off resetTransport, so must be last before return
+		t.Close(connectionErrorf(true, nil, "initial http2 frame from server is not a settings frame"))
```

Comment:
Might as well include the frame type here, as well (`....: %T", frame)`).

Reply:
done